13-1-1/recent_data.R

```diff
@@ -0,0 +1,154 @@
+# This file is called by compile_tables.R.
+# It processes the data about number of deaths caused by natural disasters in England
+# and Wales 2013-present along with the Age-Standardised mortality rate for those killed
+# by natural disasters.
+
+
+# Load in the datasets----
+
+# Includes data for all causes of death (non-disaster related)
+all_deaths <- read.csv(nomis_disaster_deaths_link) %>%
+  mutate(across(where(is.factor), as.character)) %>%
+  mutate(across(where(is.character), str_to_sentence)) %>% 
+  mutate(across(where(is.character), str_squish)) 
+
+all_mortality <- read.csv(nomis_mortality_link) %>% 
+  mutate(across(where(is.factor), as.character)) %>% 
+  mutate(across(where(is.character), str_to_sentence)) %>% 
+  mutate(across(where(is.character), str_squish)) 
+
+# Define some functions----
+
+get_GeoCode <- function(countryName){
+  if(countryName=="England") {return("E92000001")}
+  else if(countryName=="Wales") {return("W92000004")}
+  else if(countryName=="") {return("K04000001")}
+  else {return("")}
+}
+
+# Clean up all_deaths----
+
+all_deaths_cleaned <- all_deaths 
+all_deaths_cleaned <- all_deaths_cleaned %>% 
+  rename(Sex = GENDER_NAME, 
+         `Cause of death` = CAUSE_OF_DEATH_NAME, 
+         Value = OBS_VALUE, 
+         Country = GEOGRAPHY_NAME, 
+         Year = DATE) %>% 
+  mutate(`Observation status` = all_deaths$OBS_STATUS_NAME %>% unique()) %>%
+  mutate(GeoCode = "") %>% 
+  mutate(Series = "Number of deaths from exposure to forces of nature") %>% 
+  mutate(`Unit multiplier` = "") %>%
+  mutate(Units = "Number") %>% 
+  select(Year, 
+         Series, 
+         Country, 
+         Sex, 
+         `Cause of death`, 
+         `Observation status`, 
+         `Unit multiplier`, 
+         `Units`, 
+         GeoCode, 
+         Value) %>% 
+  # Keep only the rows for natural disaster deaths
+  filter(grepl("^[X][0-9][0-9]", `Cause of death`)) %>% # Note: we only match to beginning
+  # Exclude any rows corresponding to data about non EW residents
+  filter(Country %in% c("England", "Wales", "England and wales")) %>% 
+  mutate(Sex = ifelse(Sex == "Total", "", Sex)) %>%
+  # Replace "England and wales" with ""
+  mutate(Country = ifelse(Country == "England and wales", "", Country)) %>%
+  mutate(GeoCode = lapply(Country, get_GeoCode)) %>% 
```

Comment:
you can simplify this to:
`mutate(GeoCode = get_GeoCode(Country))`. You can then also get rid of the next line as it naturally returns a character string rather than a list.